File: src/java/product-api/src/main/java/com/product/api/core/HandlerResponse.java
package com.product.api.core;

import java.util.List;

public class HandlerResponse<T> {
    private final T data;
    private final List<String> message;
    private final boolean success;

    public HandlerResponse(T data, List<String> message, boolean success) {
        this.data = data;
        this.message = message;
        this.success = success;
    }

    public boolean isSuccess() {
        return success;
    }

    public List<String> getMessage() {
        return message;
    }

    public T getData() {
        return data;
    }
}

File: src/order-service/src/Product.Api/ProductApi.Core/HandlerResponse.cs
// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
// This product includes software developed at Datadog (https://www.datadoghq.com/).
// Copyright 2024 Datadog, Inc.

using System.Text.Json.Serialization;

namespace ProductApi.Core;

public record HandlerResponse<T>(T? Data, bool IsSuccess, List<string> Message)
{
    [JsonPropertyName("isSuccess")]
    public bool IsSuccess { get; init; } = IsSuccess;
    
    [JsonPropertyName("data")]
    public T? Data { get; init; } = Data;
    
    [JsonPropertyName("message")]
    public List<string> Message { get; init; } = Message;
}